### ISSUE TITLE
:const sv init

### ISSUE THREAD

**Issue by rurban** [perl11/cperl#294]
With #291 I fixed :const sv handling. Previously only const subs were made const. 
Now 
```
our %bug_tracker :const = (
    'App::Cpan'             => undef,
...
);
``` 
will fail adding to the hash at runtime. Defer the attributes->import call after the statement.
This is easier than my old approach a few years ago with private assign bits in the `typed/const` branch.

Without a fix `my $var :const` is useless, as you cannot assign any values to it.

1.  This must be done either via massive OP flags trickery (CONST_INIT ->
    {a,s}assign). See my 5 year old [typed/const branch](https://github.com/perl11/cperl/commits/typed/const) with
    STATE,CONSTINIT,CONST bits for all pad ops, PAD bit for const,
    plus OPf_SPECIAL for {a,s}assign, plus run-time checks. But it catches
    writing to such const vars at compile-time.
    commit 1fc087d778923690ff6579cbface1c2b433c6d8b 

2. Or simply just in the parser, moving the attribute->import call after
    the assignment.
    No run-time READONLY unset/set, no compile-time catches of writing to
    such a const var.

3. Both. Use the deferred attribute->import call simply in the parser to avoid checking both assign ops for the :const case at run-time. Set the var READONLY already plus a special CONST bit/change from OP_PADxV to OP_CONST to catch wrong compile-time writes, and enable constant folding and more compile-time checks and optims, like shaped arrays and types.

Done with 1e6d6978be875f65a0eeb504e2cc33c970839377 in master.

**Comment by rurban** [comment]
Now what's missing is the compile-time const for constant folding, and optimizations in `attrs_has_const()` to return the number of found attribs. 
If 1 const is the only one, and no `attribute->import(... "const")` needs to be done.
This should be merged with `attrs_runtime()` in the ffi, which decides what attrs need deferral to run-time. A single const not, a combination of const with some other non-core attrs does.

And then the assignment should be lifted from BEGIN, so that constant folding will work.
Set the value and set READONLY already at compile-time, similar to a CONSTSUB.
Currently it only catches illegal writes.

Done so with 010ef5bf558f4849dcf382bb68cd10e84167902e  :const init - fold also ranges

**Comment by rurban** [comment]
Merged to master with commit 284f6093ff1fc824821cb5e07c413a3492d98fdc
Author: Reini Urban <rurban@cpan.org>
Date:   Sun Jun 18 01:40:17 2017 +0200

    :const init - honor more attribs

    with constant folded assignments.
    otherwise just the folded op would be returned, and
    the remaining run-time attrs ignored.
